_rules/form-control-label-descriptive-cc0f0a.md

```diff
@@ -15,14 +15,16 @@ input_aspects:
   - CSS Styling
 acknowledgements:
   authors:
+    - Jean-Yves Moyen
+    - Wilco Fiers
+  previous_authors:
     - Dagfinn Rømen
     - Geir Sindre Fossøy
-    - Wilco Fiers
 ---
 
 ## Applicability
 
-This rule applies to any HTML `label` element or other element referenced by `aria-labelledby` that, is [visible][] and is programmatically associated with an HTML element that has one of the following [semantic roles][]:
+This rule applies to any HTML `label` element or other element referenced by `aria-labelledby` that is [visible][], and is programmatically associated with an HTML element which is [visible][] and has one of the following [semantic roles][semantic role]:
```

Comment:
I think we can simplify this by quite a bit (and solve #1022 in the process) by creating a new definition:

```suggestion
This rule applies to any [visible][] HTML element that is a [programatic label][] for another HTML element that has one of the following [semantic roles][semantic role]:
```

And then add this as a definition (will need some word smithing and linking out to specs):

```md
---
name: programatic label
key: programatic-label
---

Element A is a _programatic label_ of element B, if both elements are [visible][], and one of the following is true:

1. element A is a label element that is implicitly or explicitly associated with element B; or
2. Element A is referenced by ID in the `aria-labelledby` attribute of element A
```